met-web/src/components/engagement/view/EmailPanel.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import {
+    Grid,
+    Typography,
+    Checkbox,
+    Button,
+    TextField,
+    FormControl,
+    FormControlLabel,
+    FormHelperText,
+} from '@mui/material';
+import { EmailPanelProps } from './types';
+
+const style = {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: 700,
+    bgcolor: 'background.paper',
+    boxShadow: 10,
+    pt: 2,
+    px: 4,
+    pb: 3,
+    height: 700,
+};
+
+const label = { inputProps: { 'aria-label': 'Checkbox demo' } };
+
+export default function EmailPanel(props: EmailPanelProps) {
+    const [checked, setChecked] = useState(false);
+    const [termsError, setTermsError] = useState(false);
+    const [emailError, setEmailError] = useState(false);
+
+    useEffect(() => {
```

Comment:
please check error handling in EngagementForm component. This works but it will need to add a useState and a useEffect for each error. In EngagementForm there's one useState for error and one validateForm that checks for errors functions. Please consider doing something similar